controllers/cbcontainerscluster_controller.go

```diff
@@ -115,6 +159,13 @@ func (r *CBContainersClusterReconciler) Reconcile(ctx context.Context, req ctrl.
 	return ctrl.Result{Requeue: stateWasChanged}, nil
 }
 
+func (r *CBContainersClusterReconciler) apiGateway(cbContainersCluster *cbcontainersv1.CBContainersCluster, accessToken string) Gateway {
+	if r._apiGateway == nil {
```

Comment:
Im not sure if it can happen but:

if some one will change something in the cbContainersCluster or the accessToken and we already created the _apiGateway, we might return here invalid gateway

Reply:
it absolutely can. I removed the singleton. creating the gateway should not be an expensive operation. it just sets some struct fields.